### DISCUSSION TITLE
How to install builtin+ELPA packages like `eglot` or `project`

### DISCUSSION THREAD

**Discussion by darkfeline** [emacscollective/borg#153] [Support] [answered]
This is related to https://github.com/emacscollective/borg/discussions/115, but there are some packages that are builtin and updated through ELPA like `eglot` which AFAICT come directly from the Emacs repository?  I wonder if there's a (sane) way to install these using Borg.

**Comment by Thaodan** [comment, marked as answer]
There isn't any you have to do besides to assimilate the drone. Borg will setup the load-path so that  the Eglot installed through ELPA will overshadow the Eglot that is builtin.
A good place to see that is when you enter `M-x report-emacs-bug` in the very bottom it will tell you which file overshadows which.

**Comment by darkfeline** [comment]
The problem is that you need to do a full clone of the entire Emacs source repo for each builtin package, which means that your personal Emacs config is maybe five times the size of the full Emacs source repo.  I was wondering if anyone came up with a saner solution than that.

Edit: I'm talking about using Borg to install/update these instead of the builtin package.el, in case that wasn't clear.

**Comment by Thaodan** [comment]
Ah that's how you mean it. I wasn't thinking of someone using packages from Emacs Git. I don't know usually these packages have their own repository, I'm not sure how much it is intended that they are used from inside Emacs Git as they usually also depend on other features of that Emacs version.
But with Eglot that seems to be on purpose.
Borg doesn't have anything to do directly with a package being installed on it's own, you can add a repository and add many package from to your load-path.
You just had to add load-path for each directory you want to use packages from. The issue with Eglot is that it inside the regular Emacs repository along other files in the progmodes folder, for Elpa you just don't see it as there's a manual recipe to take that file and move it into a tarball.
